build.proj

```diff
@@ -33,8 +33,10 @@
     <ExeExtension>.exe</ExeExtension>
     <ExeExtension Condition=" '$(OS)' != 'Windows_NT' "></ExeExtension>
     <DotnetStage0>$(Stage0Path)/dotnet$(ExeExtension)</DotnetStage0>
-
     <DotnetCliBuildDirectory>$(RepoRoot)/build_projects/dotnet-cli-build</DotnetCliBuildDirectory>
+    <OutputDirectory>$(RepoRoot)/artifacts</OutputDirectory>
+    <PackagesDir>$(RepoRoot)/.nuget</PackagesDir>
```

Comment:
I see your concern with that and agree although the properties will come in by magic, there's less magic when you inspect the overall implementation. With the variables encapsulated in a file that describes the reason for their existence we gain clarity that I think we otherwise lose here.